controllers/C_Prescription.class.php

```diff
@@ -688,40 +707,157 @@ function multiprint_action($id = "")
             $this->function_argument_error();
         }
 
-        $pdf = new Cezpdf($GLOBALS['rx_paper_size']);
-        $pdf->ezSetMargins($GLOBALS['rx_top_margin'], $GLOBALS['rx_bottom_margin'], $GLOBALS['rx_left_margin'], $GLOBALS['rx_right_margin']);
-        $pdf->selectFont('Helvetica');
+        list($pdf, $patient) = $this->generatePdfObjectForPrescriptionIds($id);
 
-        // $print_header = true;
-        $on_this_page = 0;
+        $pFirstName = $patient->fname; //modified by epsdky for prescription filename change to include patient name and ID
+        $pFName = convert_safe_file_dir_name($pFirstName);
+        $modedFileName = "Rx_{$pFName}_{$patient->id}.pdf";
+        $pdf->ezStream(array('Content-Disposition' => $modedFileName));
+        return;
+    }
 
-        //print prescriptions body
-        $this->_state = false; // Added by Rod - see Controller.class.php
-        $ids = preg_split('/::/', substr($id, 1, strlen($id) - 2), -1, PREG_SPLIT_NO_EMPTY);
+    function multiprintplain_header($p)
+    {
+        $this->providerid = $p->provider->id;
+        $sql = "SELECT f.name, f.street, f.state, f.postal_code, f.phone, if(f.fax != '', f.fax, '') FROM users JOIN facility AS f ON f.name = users.facility where users.id = ?";
+        $result = \OpenEMR\Common\Database\QueryUtils::fetchRecords($sql, [$p->provider->id]);
+        $address = '';
+        if (!empty($result)) {
+            $res = $result[0];
+            $parts = [];
+            $parts[] = $res['name'];
+            $parts[] = $res['street'];
+            if (!empty($res['city'])) {
+                $parts[] = $res['city'] ?? '' . ', ' . $res['state'] ?? '' . ' ' . $res['postal_code'] ?? '';
+            } else if (!empty($res['state']) && !empty($res['postal_code'])) {
+                $parts[] = $res['state'] ?? '' . ' ' . $res['postal_code'] ?? '';
+            }
+            if (!empty($res['phone'])) {
+                $parts[] = xl('Tel:') . $res['phone'] ?? '';
+            }
+            if (!empty($res['fax'])) {
+                $parts[] = xl('Fax:') . $res['fax'] ?? '';
+            }
+            $address = implode("\n", $parts);
+            if (trim($address) == "") {
+                $address = "";
+            }
+        }
+        echo xl("Digital Prescription Information") . "\n";
+        echo xl("Prescriber") . "\n";
+        echo $address;
```

Comment:
is this getting echo'd into html or something else?

Reply:
Gets spit out into json in getDefaultMailClientText_action() and then is used in general_list.html.twig in lines 479-508 in the window.open call.